mlir/test/Dialect/Tensor/bubble-reshapes.mlir

```diff
@@ -20,6 +20,26 @@ func.func @bubble_parallel_reshapes(%arg0: tensor<?x?x?x?xf32>, %s0: index, %s1:
 
 // -----
 
+func.func @bubble_parallel_reshapes2(%arg0: tensor<?x2x2x6xf32>, %s0: index, %s1: index) -> tensor<?x4x2x3xf32> {
+  %c2 = arith.constant 2 : index
+  %c3 = arith.constant 3 : index
+  %collapse = tensor.collapse_shape %arg0 [[0], [1, 2], [3]] : tensor<?x2x2x6xf32> into tensor<?x4x6xf32>
+  %expand = tensor.expand_shape %collapse [[0], [1], [2, 3]]
+              output_shape [%s0, %s1, %c2, %c3] : tensor<?x4x6xf32> into tensor<?x4x2x3xf32>
```

Comment:
I'm not that familiar with this op anymore but I expected `output_shape [%s0, 4, 2, 3]`.

Reply:
Actually, I'm surprised that the verifier allows this op.

Reply:
yeah, probably missing a canonicalizer here.

Reply:
I expected the `output_shape` to match the result type. I.e., an output dim must be static iff the respective dim is static in the result type. Is that not the case?

Reply:
The verifier doesnt enforce it (would be wrong to do so... its not necessarily wrong IR), but we could convert the dynamic values in `output_shape` to static values.

Reply:
For consistency with other tensor dialect ops, I would recommend to make the verifier stricter. See discussion [here](https://discourse.llvm.org/t/tensor-ops-with-dynamic-sizes-which-behaviour-is-more-correct/82612).